front/app/api/ideas/useUpdateIdea.ts

```diff
@@ -0,0 +1,29 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import ideasKeys from './keys';
+import { IIdea, IIdeaUpdate } from './types';
+
+type IUpdateIdeaObject = {
+  id: string;
+  requestBody: IIdeaUpdate;
+};
+
+const updateIdea = ({ id, requestBody }: IUpdateIdeaObject) =>
+  fetcher<IIdea>({
+    path: `/ideas/${id}`,
+    action: 'patch',
+    body: requestBody,
+  });
+
+const useUpdateIdea = () => {
+  const queryClient = useQueryClient();
+  return useMutation<IIdea, CLErrors, IUpdateIdeaObject>({
+    mutationFn: updateIdea,
+    onSuccess: () => {
```

Comment:
There might be other things that are affected by this mutation need to be invalidated (if already part of RQ) or refetched here (streams.refetch) like idea counts or idea filter counts. Otherwise we will have data freshness problems

Reply:
That part is not very clear to me. I've created a ticket to handle this when all mutations are done if that's ok? https://citizenlab.atlassian.net/browse/CL-3018

Reply:
OK!